module/mempool/stdmap/backdata/mapBackData.go

```diff
@@ -1,145 +1,125 @@
 package backdata
 
-import (
-	"github.com/onflow/flow-go/model/flow"
-)
-
 // MapBackData implements a map-based generic memory BackData backed by a Go map.
 // Note that this implementation is NOT thread-safe, and the higher-level Backend is responsible for concurrency management.
-type MapBackData struct {
+type MapBackData[K comparable, V any] struct {
 	// NOTE: as a BackData implementation, MapBackData must be non-blocking.
 	// Concurrency management is done by overlay Backend.
-	entities map[flow.Identifier]flow.Entity
+	dataMap map[K]V
 }
 
-func NewMapBackData() *MapBackData {
-	bd := &MapBackData{
-		entities: make(map[flow.Identifier]flow.Entity),
+func NewMapBackData[K comparable, V any]() *MapBackData[K, V] {
+	bd := &MapBackData[K, V]{
+		dataMap: make(map[K]V),
 	}
 	return bd
 }
 
-// Has checks if backdata already contains the entity with the given identifier.
-func (b *MapBackData) Has(entityID flow.Identifier) bool {
-	_, exists := b.entities[entityID]
+// Has checks if a value is stored under the given key.
+func (b *MapBackData[K, V]) Has(key K) bool {
+	_, exists := b.dataMap[key]
 	return exists
 }
 
-// Add adds the given entity to the backdata.
-func (b *MapBackData) Add(entityID flow.Identifier, entity flow.Entity) bool {
-	_, exists := b.entities[entityID]
+// Add attempts to add the given value to the backdata, without overwriting existing data.
+// If a value is already stored under the input key, Add is a no-op and returns false.
+// If no value is stored under the input key, Add adds the value and returns true.
+func (b *MapBackData[K, V]) Add(key K, value V) bool {
+	_, exists := b.dataMap[key]
 	if exists {
 		return false
 	}
-	b.entities[entityID] = entity
+	b.dataMap[key] = value
 	return true
 }
 
-// Remove removes the entity with the given identifier.
-func (b *MapBackData) Remove(entityID flow.Identifier) (flow.Entity, bool) {
-	entity, exists := b.entities[entityID]
-	if !exists {
-		return nil, false
+// Remove removes the value with the given key.
+// If the key-value pair exists, returns the value and true.
+// Otherwise, returns the zero value for type V and false.
+func (b *MapBackData[K, V]) Remove(key K) (value V, ok bool) {
+	value, ok = b.dataMap[key]
+	if !ok {
+		return value, false
 	}
-	delete(b.entities, entityID)
-	return entity, true
+	delete(b.dataMap, key)
+	return value, true
 }
 
-// Adjust adjusts the entity using the given function if the given identifier can be found.
-// Returns a bool which indicates whether the entity was updated as well as the updated entity.
-func (b *MapBackData) Adjust(entityID flow.Identifier, f func(flow.Entity) flow.Entity) (flow.Entity, bool) {
-	entity, ok := b.entities[entityID]
+// Adjust adjusts the value using the given function if the given key can be found.
+// It returns the updated value along with a boolean indicating whether an update occurred.
```

Comment:
```suggestion
// Returns:
//   - the updated value for the key (if the key exists)
//   - a boolean indicating whether the key was found (and the update applied)
```
Adjusting wording slightly here to reflect that we return true even if the update is a no-op.